gulp.config.js

```diff
@@ -81,12 +42,109 @@ module.exports = function() {
             // Seems to break sourcemaps, so conditionally minify css in gulpfile
             // outputStyle: 'compressed',
             precision: 8,
+            includePaths: [src + theme, src + base]
         },
         uglifySettings: {
             mangle: false,
         },
     };
 
+    setConfigPaths();
     return config;
+
+    //////////////////////////////////
+
+    // Fill the config with themes paths
+    function setConfigPaths() {
+        addThemePaths('./' + src + config.base + '/');
+        addThemePaths('./' + src + config.theme + '/');
+
+        // Styles root
+        var themeStylesRoot = './' + src + config.theme + '/app.scss';
+        var match = glob.sync(themeStylesRoot);
+        config.styles.root = (match.length) ? themeStylesRoot : './' + src + config.base + '/app.scss';
+
+        return config;
+    }
+
+    // Adds paths of a single theme to the config
+    function addThemePaths(themeDir) {
+        // Theme paths patterns
+        var themePaths = {
+            fonts: themeDir + '_fonts/**/*.{otf,svg,eot,ttf,woff,woff2}',
+            html: {
+                root: themeDir + '*.html',
+                nonRoot: [
+                    themeDir + '**/*.html',
+                    '!' + themeDir + email + '**/*',
+                    '!' + themeDir + '*.html',
+                ],
+            },
+            media: [
+                // Ignore media, declaring the _images folder directly moves all of its contents
+                // without the container folder getting in the way
+                themeDir + '_images/**/*.{png,gif,jpg,jpeg,ico,svg,mp4,ogv,webm,pdf}',
+                themeDir + '**/*.{png,gif,jpg,jpeg,ico,svg,mp4,ogv,webm,pdf}',
+                '!' + themeDir + '_fonts/*',
+            ],
+            misc: [
+                themeDir + '*.{htaccess,ico,xml}',
+                themeDir + 'humans.txt'
+            ],
+            robots: {
+                default: themeDir + 'robots.dev.txt',
+                prod: themeDir + 'robots.prod.txt',
+            },
+            styles: {
+                all: [
+                    themeDir + '**/*.scss',
+                    themeDir + '**/*.css',
+                ],
+            },
+            scripts: {
+                app: [
+                    temp + 'config.js', // this is a built file
+                    themeDir + '**/init.js',
+                    themeDir + '**/*.js',
+                    '!' + themeDir + '_lib/**/*', // don't clobber lib
+                ],
+                lib: [
+                    themeDir + '_lib/jquery.min.js',
+                    themeDir + '_lib/angular.min.js',
+                    themeDir + '_lib/**/*.min.js',
+                ],
+            },
+        };
+
+        mergeConfigPaths(config, themePaths);
+    }
+
+    // Adds to the config new paths
+    // iterates over newConfig path keys and adds paths to the current config
+    function mergeConfigPaths(currConfig, newConfig) {
```

Comment:
What is the purpose of the `config` object?  

With the name `mergeConfigPaths`, the part I'm missing is what kind of configuration data I'm working with.  Perhaps `mergeThemeConfigPaths(currentThemeConfig, newThemeConfig)` is longer, but gets us closer to removing the comment above it.